### PR TITLE
`yarn <command> ls` command cleanup

### PR DESCRIPTION
**Pull request by tpina**
**Summary**

From https://github.com/yarnpkg/yarn/pull/3898#issuecomment-318609494

This PR is to clean up the remaining undeprecated `yarn <command> ls` command and update the usage template.

**Test plan**

Some rounds of functional manual testing to assert that the correct messages are displayed and previous commands work as per expected. Touched files do not have any associated tests.

### CODE REVIEW

File: src/cli/commands/team.js [hunk 1]
```diff
@@ -38,14 +43,19 @@ function explodeScopeTeam(arg: string, requireTeam: boolean, reporter: Reporter)
   };
 }
 
-function warnDeprecation(reporter: Reporter) {
-  reporter.warn(`\`yarn team rm\` is deprecated. Please use \`yarn team remove\`.`);
+function warnDeprecation(reporter: Reporter, deprecationWarning: DeprecationWarning) {
+  reporter.warn(`\`yarn team ${deprecationWarning.deprecatedCommand}\` is deprecated.
```
Comment (BYK) [inline src/cli/commands/team.js]: You should be using `reporter.lang` and add a language key for these so they can be translated.

File: src/cli/commands/team.js [hunk 2]
```diff
@@ -38,14 +43,19 @@ function explodeScopeTeam(arg: string, requireTeam: boolean, reporter: Reporter)
...
+  Please use \`yarn team ${deprecationWarning.currentComand}\`.`);
 }
 
-function wrapRequired(callback: CLIFunctionWithParts, requireTeam: boolean, isDeprecated?: boolean): CLIFunction {
+function wrapRequired(
+  callback: CLIFunctionWithParts,
+  requireTeam: boolean,
+  subCommandDeprecated?: DeprecationWarning,
```
Comment (BYK) [inline src/cli/commands/team.js]: I'd rename this to `deprecationMessage` or `deprecationInfo`

**Comment by tpina** [comment]
@BYK  and @voxsim 

Done. Awaiting for feedback.

File: src/cli/commands/team.js [hunk 3]
```diff
@@ -71,23 +81,31 @@ function wrapRequired(callback: CLIFunctionWithParts, requireTeam: boolean, isDe
...
+): CLIFunction {
+  return wrapRequired(
+    function(
+      parts: TeamParts,
+      config: Config,
+      reporter: Reporter,
+      flags: Object,
+      args: Array<string>,
+    ): CLIFunctionReturn {
+      if (args.length === 1) {
```
Comment (BYK) [inline src/cli/commands/team.js]:
```
return args.length === 1 && callback(...);
```

Also would be nice to explain why we do this only when argument count is exactly 1.

**Comment by BYK** [comment]
@tpina thanks a lot! This looks good to me. Would you like to address my changes or just merge it as it is?

**Comment by tpina** [comment]
@BYK  I'll address your changes/comments. Let me take another look at it.

File: src/cli/commands/team.js (hunk 3 above)
Comment (tpina) [inline src/cli/commands/team.js]: `yarn team list` only takes one arg which is either: `<scope>` or `<scope:team>` more than one arg and the command should fail.

File: src/cli/commands/team.js (hunk 2 above)
Comment (tpina) [inline src/cli/commands/team.js]: Will do

File: src/cli/commands/team.js (hunk 1 above)
Comment (tpina) [inline src/cli/commands/team.js]: Ok not a problem!

**Comment by tpina** [comment]
@BYK  a really annoying thing with the `reporter.lang` - it keeps putting quote on the params so the final string reads: `warning yarn "team" "ls" is deprecated. Please use `yarn "team" "list"`.

Is there any way around this?

**Comment by tpina** [comment]
meh... workaround added. See if you agree once checks have completed.

File: src/reporters/lang/en.js [hunk 4]
```diff
@@ -324,6 +324,8 @@ const messages = {
     'Installing Yarn via Yarn will result in you having two separate versions of Yarn installed at the same time, which is not recommended. To update Yarn please follow https://yarnpkg.com/en/docs/install .',
 
   scopeNotValid: 'The specified scope is not valid.',
+
+  yarnDeprecatedCommand: '`yarn $0 $1` is deprecated.\n Please use `yarn $0 $2`.',
```
Comment (BYK) [inline src/reporters/lang/en.js]: This would repeat the `yarn` part in the message :(

I'll fix it and then merge.

**Comment by tpina** [comment]
@BYK you're right. Apologies for that.

**Comment by Lucas-C** [comment]
I think the current doc does not reflect the `ls` -> `list` change:
https://yarnpkg.com/en/docs/cli/owner

**Comment by BYK** [comment]
@Lucas-C thanks for the heads up! Filed https://github.com/yarnpkg/website/issues/660

**Pull request merged**